baton/modules/twilio/twilio.go

```diff
@@ -0,0 +1,168 @@
+package twilio
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/gorilla/mux"
+)
+
+type Twilio struct {
+	UserId, ApiKey string
+}
+
+var (
+	smsCallbacks  = make([]callback, 0)
+	callCallbacks = make([]callback, 0)
+)
+
+type callback struct {
+	number string
+	resp   chan<- interface{}
+	data   string
+}
+
+var client = &http.Client{}
+
+func (t Twilio) RunCommand(cmd string, body interface{}, resp chan<- interface{}) error {
+	newBody := body.(map[string]interface{})
+	switch cmd {
+	case "send-sms":
+		return t.sendSMS(newBody, resp)
+	case "recieve-sms":
+		return t.recieveSMS(newBody, resp)
+	case "send-call":
+		return t.makeCall(newBody, resp)
+	case "recieve-call":
+		return t.recieveCall(newBody, resp)
+	default:
+		return errors.New("unknown command: " + cmd)
+	}
+}
+
+func (t Twilio) sendSMS(body map[string]interface{}, resp chan<- interface{}) error {
+	to := body["to"].(string)
+	from := body["from"].(string)
+	message := body["body"].(string)
+	form := url.Values{"To": {to}, "From": {from}, "Body": {message}}
+
+	res, err := t.postForm(fmt.Sprintf("https://api.twilio.com/2010-04-01/Accounts/%s/Messages.json", t.UserId), form)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+
+	var out map[string]interface{}
+	if err := json.NewDecoder(res.Body).Decode(&out); err != nil {
+		return err
+	}
+	delete(out, "account_sid")
+	resp <- out
+	return nil
+}
+
+func (t Twilio) makeCall(body map[string]interface{}, resp chan<- interface{}) error {
+	to := body["to"].(string)
+	from := body["from"].(string)
+	twiml := body["twiml"].(string)
+	form := url.Values{"To": {to}, "From": {from}, "Url": {"http://524b95fe.ngrok.io/baton/webhooks/Twilio/call"}} //temporary
```

Comment:
As mentioned previously and on #16, this will be solved with configuration. The recent change was due to the merge of #16 which changed the path from `/webhooks/...` to `/baton/webhooks/...`